### DISCUSSION TITLE
Only abstract entities are discovered after upgrading MikroORM version

### DISCUSSION THREAD

**Discussion by ropbastos** [mikro-orm/mikro-orm#5854] [Q&A]
### Describe the bug

Upgrading the following packages from 6.2.10-dev.72 to 6.2.10-dev.73 (or to the newest release, `6.3.0`) causes the an error when I run `APP_ENV=local MIKRO_ORM_CLI_USE_TS_NODE=true npx mikro-orm migration:up`.

The error:
```
MetadataError: Only abstract entities were discovered, maybe you forgot to use @Entity() decorator?
```

The packages:
```
    "@mikro-orm/cli": "6.2.10-dev.73",
    "@mikro-orm/core": "6.2.10-dev.73",
    "@mikro-orm/migrations": "6.2.10-dev.73",
    "@mikro-orm/postgresql": "6.2.10-dev.73",
    "@mikro-orm/reflection": "6.2.10-dev.73",
    "@mikro-orm/seeder": "6.2.10-dev.73",
 ```
 

-  Figured it might be easy to understand what goes on since it starts happening in at 6.2.10-dev.73. Will post a reproduction as soon as I have time. Appreciate any help 🙏

### Reproduction

//TODO

### What driver are you using?

@mikro-orm/postgresql

### MikroORM version

6.3.0

### Node.js version

v20.15.1

### Operating system

macOS Sonoma 14.5

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/mikro-orm/mikro-orm/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://mikro-orm.io/docs).
- [X] Check that there isn't [already an issue](https://github.com/mikro-orm/mikro-orm/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/mikro-orm/mikro-orm/discussions) or join our [Discord](https://discord.gg/w8bjxFHS7X).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by B4nan** [comment]
Please don't open issues without any reproduction.

**Comment by ahhmarr** [comment]
I'm encountering somewhat same issue

Note: this seems to be only when I'm using CLI and not via the code 

repo to reproduce

https://github.com/ahhmarr/mikro-orm-enitities-issue

My issue is that I cannot run the seeder inside docker container

throws the same mentioned error  
```
MetadataError: Only abstract entities were discovered, maybe you forgot to use @Entity() decorator?
```

**Comment by tsvetann** [comment]
Just upgraded and had bunch of issues and that is one of them. Any ideas?

Looks like the following check fails for me:
`discovered.filter(meta => meta.name).length === 0`

There are discovered entities but none of them have the `name` property.

**Reply by B4nan** [reply]
> Any ideas?

Any reproduction?

**Comment by thdk** [comment]
@tsvetann @ahhmarr This error can happen if you have multiple versions of any mikro-orm package in your dependency graph.

You can check this with:

```sh
npm ls @mikro-orm/core # or any other package that your are using from mikro-orm 

```

Example how it looks **after** I fixed my dependency versions:

```sh
# npm ls @mikro-orm/postgresql
@org/platform@0.0.1-never-publish /Users/thdk/repos/repo
├─┬ @org/api@1.134.5 -> ./apps/api
│ ├─┬ @org/lib-1@8.1.1
│ │ └── @mikro-orm/postgresql@6.3.8 deduped
│ ├─┬ @org/lib-2@6.1.0
│ │ └─┬ @org/lib-1@8.2.0
│ │   └── @mikro-orm/postgresql@6.3.8 deduped
│ └── @mikro-orm/postgresql@6.3.8
└─┬ @org/lib-3@3.0.2 -> ./packages/lib-3
  └── @mikro-orm/postgresql@6.3.8 deduped
```

I had the same error and it was resolved after I've updated all my packages to the **same** version.

**Reply by B4nan** [reply]
I've modified the error message to also mention this: 3835e4009bc8ba5fcffb0857450572068a38fe3f

We actually do have a validation for this, but it's apparently not working in monorepos.

https://github.com/mikro-orm/mikro-orm/blob/master/packages/core/src/utils/ConfigurationLoader.ts#L318

**Reply by tsvetann** [reply]
maybe we can try something like `npm find-dupes`?

**Reply by B4nan** [reply]
I don't want to call NPM commands from inside the library, people might be using different package manager.

**Reply by tsvetann** [reply]
Agree. then how about a third party tool like `find-duplicate-dependencies`. I see there are quite a few alternatives

**Reply by B4nan** [reply]
I'll take a look for inspiration, I would rather improve the implementation we already have instead of adding more dependencies.
